### PR TITLE
feat: Added the Request Certificate Alert

### PR DESCRIPTION
**Pull request by staubina**
[MICROBA-678]

    When a certificate is in an unexpected state (i.e. notpassing with a
    passing grade) this alert will allow the user to attempt to resolve the
    issue on their own. It will run the code that checks the certificates
    status. It requires that the course is configured to allow users to
    Request Certificates though.

Screen Shot 2021-07-13 at 9 05 33 AM
Screen Shot 2021-07-13 at 9 05 44 AM

### CODE REVIEW

File: src/course-home/outline-tab/alerts/certificate-status-alert/CertificateStatusAlert.jsx [hunk 1]
```diff
@@ -7,23 +7,29 @@ import {
...
 } from '@edx/frontend-platform/i18n';
 import { Alert, Button } from '@edx/paragon';
+import { useDispatch } from 'react-redux';
+
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faCheckCircle, faExclamationTriangle } from '@fortawesome/free-solid-svg-icons';
 import { getConfig } from '@edx/frontend-platform';
 import certMessages from './messages';
 import certStatusMessages from '../../../progress-tab/certificate-status/messages';
+import { requestCert } from '../../../data/thunks';
```
Comment (oliviaruizknott) [inline src/course-home/outline-tab/alerts/certificate-status-alert/CertificateStatusAlert.jsx:17]: Our bot is calling this out; is this used?

File: src/course-home/outline-tab/alerts/certificate-status-alert/CertificateStatusAlert.jsx [hunk 2]
```diff
@@ -7,23 +7,29 @@ import {
...
 
 export const CERT_STATUS_TYPE = {
   EARNED_NOT_AVAILABLE: 'earned_but_not_available',
   DOWNLOADABLE: 'downloadable',
+  REQUESTING: 'requesting',
   UNVERIFIED: 'unverified',
 };
 
 function CertificateStatusAlert({ intl, payload }) {
+  const dispatch = useDispatch();
```
Comment (oliviaruizknott) [inline src/course-home/outline-tab/alerts/certificate-status-alert/CertificateStatusAlert.jsx:27]: Our bot is calling this out; is this used?

File: src/course-home/outline-tab/alerts/certificate-status-alert/CertificateStatusAlert.jsx [hunk 3]
```diff
@@ -7,23 +7,29 @@ import {
...
   const {
     certificateAvailableDate,
-    certStatusType,
+    certStatus,
     courseEndDate,
+    courseId,
```
Comment (oliviaruizknott) [inline src/course-home/outline-tab/alerts/certificate-status-alert/CertificateStatusAlert.jsx:32]: ... same 😄

File: src/course-home/outline-tab/alerts/certificate-status-alert/hooks.js [hunk 4]
```diff
@@ -51,14 +49,15 @@ function useCertificateStatusAlert(courseId) {
...
 
   // Only show if there is a known cert status that we want provide status on.
-  const isVisible = isEnrolled && hasCertStatus;
+  const isVisible = isEnrolled && hasAlertingCertStatus;
   const payload = {
     certificateAvailableDate,
     certURL,
-    certStatusType,
+    certStatus,
+    courseId,
```
Comment (oliviaruizknott) [inline src/course-home/outline-tab/alerts/certificate-status-alert/hooks.js:60]: Looks like this might not be necessary in the `CertificateStatusAlert`?

**Review by oliviaruizknott — COMMENTED** [review]
I’m not sure you wanted a review yet since this is a draft! And all my comments are useless because the robot already caught them. 🤖

File: src/course-home/outline-tab/alerts/certificate-status-alert/CertificateStatusAlert.jsx (hunk 1 above)
Comment (staubina) [reply src/course-home/outline-tab/alerts/certificate-status-alert/CertificateStatusAlert.jsx:17]: @oliviaruizknott As I mentioned in the description this was not up for review yet. I was simply sharing the refactor code. It was still a WIP and these were not fully used yet.

**Review by Tj-Tracy — APPROVED** [review]

**Pull request merged**